### DISCUSSION TITLE
[RFC] Унифицировать названия "базовых" компонентов

### DISCUSSION THREAD

**Discussion by eugpoloz** [VKCOM/VKUI#2362] [Ideas]
У нас понемногу увеличивается количество "базовых" компонентов, но названия у них разные. Что есть сейчас:

- **Simple**Component: `SimpleCell`, `SimpleCheckbox` (unstable)
- Component**Base**: `ModalCardBase`

Я бы предложила подумать, какой вариант мы оставляем, и переименовала в [v5](https://github.com/VKCOM/VKUI/milestone/2) компонент(-ы) соответственно.

Мне больше нравится ComponentBase, потому что тогда компоненты будут лежать рядышком друг с другом и в них проще будет ориентироваться. С другой стороны — SimpleCell в либе уже очень-очень давно и много где используется, менять для него название может быть болезненно.

**Comment by stoope** [comment]
Я за ComponentBase, так будет проще искать и ориентироваться в компонентах т.к. глаз сразу зацепляется за название компонента

**Comment by inomdzhon** [comment]
Плюсую к Стёпе, ComponentBase 👍

**Comment by NekR** [comment]
Как в таком случае будет SimpleCell называться, BaseCell или CellBase?

**Comment by inomdzhon** [comment]
> Как в таком случае будет SimpleCell называться, BaseCell или CellBase?

`CellBase`

**Comment by NekR** [comment]
У текущего нейминга есть прямое назначение. Base звучит как базовый компонент для чего-то, на основе которого строятся более сложные компоненты. У нас в доках это формулируется как "низкоуровневый компонент".

При этом, Simple или Mini компоненты это не базовые/низкоуровневые компоненты для других, а именно отдельные компоненты, упрощенные или уменьшенные соответственно.

Пример: SimpleCell и RichCell оба могли бы быть основаны на CellBase.

Если где-то есть компоненты не соответствующие этой логике, то тогда лучше их поправить.

**Comment by eugpoloz** [comment]
@NekR 

> При этом, Simple или Mini компоненты это не базовые/низкоуровневые компоненты для других, а именно отдельные компоненты, упрощенные или уменьшенные соответственно.
>
> Пример: SimpleCell и RichCell оба могли бы быть основаны на CellBase.

Сейчас `Cell` и `CellButton` основываются на `SimpleCell`, поэтому он меня и смущает. 😄 

`SimpleCheckbox` изначально тоже задумывался как штука, на базе которой потом можно будет переделать `Checkbox`, чтобы по два раза код не копировать.

**Comment by NekR** [comment]
> Сейчас Cell и CellButton основываются на SimpleCell, поэтому он меня и смущает. 😄

Это хороший поинт кстати. Вообще, какое-то время Cell и SimpleCell были одним и тем же. То есть алиас. Возможно он так просто реализован, что один компонент в другом?

Но возможно да, стоит найти базовый компонент для всех и вынести туда общую логику.
